### PR TITLE
[Clean Code]Remove SkipFailbackWrapperException and unusable code about it

### PR DESCRIPTION
**Pull request by LiZhenNet**
## What is the purpose of the change

Remove SkipFailbackWrapperException and unusable code about it

## Brief changelog

Remove SkipFailbackWrapperException and unusable code about it

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by lovepoem** [comment]
Please add some test case to verify the changes

**Comment by LiZhenNet** [comment]
> Please add some test case to verify the changes

It will test by FailbackRegistryTest .  SkipFailbackWrapperException  does not throw  in source code.
so .my change effect nothing

**Comment by chenlushun** [comment]
I think it's still reserved.

**Comment by LiZhenNet** [comment]
> I think it's still reserved.

Entities should not be multiplied unnecessarily

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/FailbackRegistry.java [hunk 1]
```diff
@@ -230,22 +230,15 @@ public void register(URL url) {
...
             boolean check = getUrl().getParameter(Constants.CHECK_KEY, true)
                     && url.getParameter(Constants.CHECK_KEY, true)
                     && !Constants.CONSUMER_PROTOCOL.equals(url.getProtocol());
-            boolean skipFailback = t instanceof SkipFailbackWrapperException;
-            if (check || skipFailback) {
-                if (skipFailback) {
-                    t = t.getCause();
-                }
-                throw new IllegalStateException("Failed to register " + url + " to registry " + getUrl().getAddress() + ", cause: " + t.getMessage(), t);
+            if (check ) {
```
Comment (satansk) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/FailbackRegistry.java]: There is an extra space here:

```Java
if (check )
```

**Comment by beiwei30** [comment]
I don't think we can remove SkipFailbackWrapperException directly. It is a **contract** for FailbackRegistry's implemenations to skip fallback, even though the existing implementations have not used it yet.

Due to the reason given above, I close this pull request. Feel free to re-open it if you don't agree.

**Pull request closed without merging**